src/ucx_van.h

```diff
@@ -800,13 +802,31 @@ class UCXVan : public Van {
   int Bind(Node &node, int max_retry) override {
     contexts_.reserve(node.num_ports);
 
+    int num_cpu_dev   = GetEnv("DMLC_NUM_CPU_DEV", 1);
+    int num_gpu_dev   = GetEnv("DMLC_NUM_GPU_DEV", 0);
+
```

Comment:
How do we typically set NUM_PORTS and DMLC_NUM_CPU_DEV/DMLC_NUM_GPU_DEV? Like
DMLC_NUM_GPU_DEV=NUM_PORTS=4 for 4 NICs?

Reply:
they should satisfy NUM_PORTS = DMLC_NUM_GPU_DEV + DMLC_NUM_CPU_DEV, and for the moment we assume there's only one type of devices

Reply:
will you need to support mixed types of device? if yes, why would need to maintain such restriction?

Reply:
yes we do want mixed types. this restriction is because the code doesn't support mixed dev types yet

Reply:
Can we remove this restriction and test it again? @pleasantrabbit

Reply:
done. I removed the the check.